universal-login-react/src/ui/commons/Balance.tsx

```diff
@@ -3,17 +3,33 @@ import './../styles/balance.sass';
 import './../styles/balanceDefault.sass';
 import {getStyleForTopLevelComponent} from '../../core/utils/getStyleForTopLevelComponent';
 import {formatCurrency} from '../../core/utils/formatCurrency';
+import Spinner from './Spinner';
 
 export interface BalanceProps {
-  amount: string;
+  amount: number | undefined;
   className?: string;
 }
 
-export const Balance = ({amount, className}: BalanceProps) => (
-  <div className="universal-login-balance">
-    <div className={getStyleForTopLevelComponent(className)}>
-      <p className="universal-login-balance-text">Balance</p>
-      <p className="universal-login-balance-amount">{formatCurrency(amount)}</p>
-    </div>
-  </div>
-);
+export const Balance = ({amount, className}: BalanceProps) => {
+
+  if (amount === undefined) {
+    return (
+      <div className="universal-login-balance">
+        <div className={getStyleForTopLevelComponent(className)}>
+          <div className="balance-spinner-wrapper">
+            <Spinner className="balance-spinner" />
+          </div>
+        </div>
+      </div>
+    );
+  } else {
+    return (
+      <div className="universal-login-balance">
+        <div className={getStyleForTopLevelComponent(className)}>
+          <p className="balance-text">Balance</p>
+          <p className="balance-amount">{formatCurrency(amount.toString())}</p>
+        </div>
+      </div>
+    );
```

Comment:
Two outer div's are identical in both branches, make a single return with `? :` operator in the middle:
```
 <div className="universal-login-balance">
    <div className={getStyleForTopLevelComponent(className)}>
{.... ? (
          <div className="balance-spinner-wrapper">
           <Spinner className="balance-spinner" />
          </div>
    ) : (
<>
 <p className="balance-text">Balance</p>
          <p className="balance-amount">{formatCurrency(amount.toString())}</p>
</>
)}
        </div>
      </div>
```

Reply:
We can extract renderBalance helper function: 
```
<div className="universal-login-balance">
    <div className={getStyleForTopLevelComponent(className)}>
        {renderBalance(amount)}
    </div>
</div>
```
```
const renderBalance = (amount:address| undefined) => {
    amount === undefined ? (
        <div className="balance-spinner-wrapper">
           <Spinner className="balance-spinner" />
        </div>
     ) : (
        <>
            <p className="balance-text">Balance</p>
            <p className="balance-amount">{formatCurrency(amount.toString())}</p>
       </>
}
```